google/cloud/internal/populate_rest_options.cc

```diff
@@ -0,0 +1,48 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/internal/populate_rest_options.h"
+#include "google/cloud/credentials.h"
+#include "google/cloud/internal/absl_str_cat_quiet.h"
+#include "google/cloud/internal/populate_common_options.h"
+#include "google/cloud/internal/rest_options.h"
+#include "absl/strings/match.h"
+
+namespace google {
+namespace cloud {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace internal {
+
+Options PopulateRestOptions(Options opts) {
+  if (!opts.has<UnifiedCredentialsOption>()) {
+    opts.set<UnifiedCredentialsOption>(
+        MakeGoogleDefaultCredentials(internal::MakeAuthOptions(opts)));
+  }
+  if (!opts.has<rest_internal::LongrunningEndpointOption>()) {
+    opts.set<rest_internal::LongrunningEndpointOption>(
+        "https://longrunning.googleapis.com");
+  }
+  if (opts.has<EndpointOption>()) {
+    std::string endpoint = opts.get<EndpointOption>();
+    if (!absl::StartsWithIgnoreCase(endpoint, "http")) {
+      opts.set<EndpointOption>(absl::StrCat("https://", endpoint));
+    }
```

Comment:
Nit: I know this is only capturing existing behavior, but starting with "http" is not the same thing as starting with "http://" or "https://".

Reply:
Ack, I am going to defer addressing this comment in this PR.

Reply:
#13397